src/errors.rs

```diff
@@ -122,10 +132,16 @@ impl std::fmt::Display for Error {
             }
             InvalidOsStr => write!(f, "Paths must be valid unicode on this platform"),
             NullInStr(_err) => write!(f, "Paths cannot include null bytes"),
-            CouldNotCheckNAtoms(_err) => write!(
+            CouldNotCheckNAtoms(_err) => {
+                write!(f, "Failed to read number of atoms in trajectory file")
+            }
+            StepSizeOutOfRange(n) => write!(f, "Step {} does not fit in usize on this platform", n),
+            NumericCastFailed { value, task, .. } => write!(
                 f,
-                "Failed to read number of atoms in trajectory file"
-            )
+                "Numeric cast from {value} failed while {task}",
```

Comment:
I think this description is too generic. Sth. like "Failed to cast number of atoms into i32" gives a better description of what went wrong and one immediately knows what the problem is without having to look into the source of this lib.